### DISCUSSION TITLE
Splitting up the block lesson in the ruby track.

### DISCUSSION THREAD

**Discussion by linkonsat** [TheOdinProject/curriculum#25448] [Ideas]
I noticed that in the [blocks lesson](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/blocks) there are 14 different knowledge checks. I think this is a good indication that e
the lesson needs to be broken up into 2 different lessons . If there are that many knowledge checks then the lesson itself is probably covering a bit more than needed as in the [knowledge check wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) it says most lessons will have 3-7. I think that the block and yield sections of the lesson could go in its own lesson with small exercises utilizing blocks. This would be followed by the proc and lambda lesson with small exercises in each to familiarize learners with how to utilize a block/yield and exercises for the other lesson using proc/lambda. It seems to be the case that using yield is a pain point for some individuals and utilizing exercises to explain it could help the lesson outcome goals stick better. Linking this other issue since this would also benefit this other issue #23481

**Comment by CouchofTomato** [comment]
@KevinMulhern 

Is this something you think can be done now or should be wait for the path review?

**Comment by linkonsat** [comment]
I have given this a bit of thought and given that @CouchofTomato mentioned there will be a path review and there is a OOP section rework issue. I think it would make more sense in my opinion to include this in a broader path upgrade. The reason being is that it makes more sense to me to upgrade and release an overall broader update to avoid scattered updates. I _think_ this would help improve the flow in having a major release at once since we could rework lessons in tandem and hopefully release a major path update with better flow as they could all be reviewed at once.

On that note would you both be open to the idea of releasing a form and pinging ruby folks to get their feedback on pain points in the ruby section? Then at least we could incorporate much more feedback for a larger ruby section update rather than it being when people are absolutely confused and stuck and frustrated and then they raise an issue. It could also give us an opportunity to see any insights learners may have gained that they simply have not shared yet.

**Comment by CouchofTomato** [comment]
Yeah that's a good idea @linkonsat
